File: Sum of Absolutes/sum_of_absolutes.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {

    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int N, Q, L, R;
    scanf("%d %d", &N, &Q);
    int *A = (int *)malloc(N * sizeof(int));
    int i, q;
    int odd = 0;
    for(i=0; i< N; i++)
        scanf("%d", A+i);
    for(q=0; q< Q; q++)
    {   
        odd = 0;
        scanf("%d %d", &L, &R);
        for(i = L-1; i < R; i++)
            odd ^= (A[i] & 1);
        printf("%s\n", (odd == 1? "Odd": "Even"));
    }
    return 0;
}

File: Sum of Absolutes/sum_of_absolutes.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
N, Q = map(int, raw_input().split())
A = map(int, raw_input().split())
scanned = []

for i in xrange(N):
	scanned.append([(A[i]&1)])
	#print "i = ", i, len(scanned)
	for j in xrange(i+1,N):
		#print "\t j = ", j, len(scanned[i])
		scanned[i].append(scanned[i][j-i-1] ^ ( A[j] & 1 ))

print scanned
for q in xrange(Q):
    L, R = map(int, raw_input().split())
    #print L-1, R-L-1
    odd = scanned[L-1][R-L]
    print (1-odd)*"Even" + odd*"Odd"
